### ISSUE TITLE
Channel selection not working

### ISSUE THREAD

**Issue by redstonedesigner** [Voxel-Fox-Ltd/Novus#14]
### Summary

In a slash command, when using discord.VoiceChannel or discord.TextChannel (could be more), it is unable to retrieve the guild ID for some reason.

### Reproduction Steps

Use a type-hint of `discord.VoiceChannel` for a parameter in a slash command.
Execute slash command.

### Minimal Reproducible Code

```python
    @commands.group(invoke_without_command=True)
    async def lobby(self, ctx: vbu.Context):
        """
        Placeholder method
        :param ctx: Command context.
        :return:
        """

    @lobby.command()
    @commands.defer()
    async def test(self, ctx: vbu.Context):
        await respond(ctx, msg=str(ctx.guild.id))

    @lobby.command()
    @commands.defer()
    @commands.guild_only()
    async def create(self, ctx: vbu.Context, channel: discord.VoiceChannel):
        pass
```

### Expected Results

No errors to be thrown and command to execute normally.

### Actual Results

Command threw an error which can be viewed here - https://hastebin.com/abahahamem.sql

### Intents

Using vbu, all intents are enabled

### System Information

- Python v3.9.7-final
- Novus v0.0.3-final
- aiohttp v3.7.4.post0
- system info: Windows 10 10.0.19043

### Checklist

- [X] I have searched the open issues for duplicates.
- [X] I have shown the entire traceback, if possible.
- [X] I have removed my token from display, if visible.

### Additional Context

This is the same setup, to my knowledge, as the one used in Apple.Py (https://github.com/Voxel-Fox-Ltd/Apple.Py/blob/11d9de7c708a0fc08e1ed90307493113fa8d25ac/cogs/vc_admin.py#L20) which I used as a reference when creating this command.

**Comment by 4Kaylum** [comment]
I can't recreate this error on master - could you confirm if this is fixed for you if you update?

**Comment by redstonedesigner** [comment]
This issue is now resolved on the master branch, however does persist in the released version `0.0.3`.
